src/client/BuilderClient.ts

```diff
@@ -0,0 +1,130 @@
+import axios, { AxiosInstance } from 'axios'
+import FormData from 'form-data'
+import { Authenticator, AuthIdentity } from 'dcl-crypto'
+import { RemoteItem, LocalItem } from '../item/types'
+import { ClientError } from './BuilderClient.errors'
+import { ServerResponse } from './types'
+
+export class BuilderClient {
+  private axios: AxiosInstance
+  private readonly AUTH_CHAIN_HEADER_PREFIX = 'x-identity-auth-chain-'
+
+  constructor(url: string, private identity: AuthIdentity) {
+    this.axios = axios.create({
+      baseURL: url
+    })
+
+    // Uses the interceptors API to add the auth headers to all requests
+    this.axios.interceptors.request.use((config) => {
+      if (!config.method || !config.url) {
+        return config
+      }
+
+      const headers = this.createAuthHeaders(
+        config.method,
+        config.url.replace(/\/v[0-9]/, '')
+      )
+      config.headers = { ...config.headers, ...headers }
+
+      return config
+    })
+
+    // Uses the interceptors API to throw when the builder server returns a 200 with a ok property set to false
+    this.axios.interceptors.response.use((response) => {
+      if (response.data && response.data.ok === false) {
+        throw new ClientError(
+          response.data.error!,
+          response.status,
+          response.data.data
+        )
+      }
+      return response
+    })
+  }
+
+  /**
+   * Creates the authorization headers for the given method and path.
+   * @param method - The HTTP method.
+   * @param path - The HTTP request path.
+   */
+  private createAuthHeaders(
+    method: string,
+    path: string
+  ): Record<string, string> {
+    const headers: Record<string, string> = {}
+    const endpoint = (method + ':' + path).toLowerCase()
+    const authChain = Authenticator.signPayload(this.identity, endpoint)
+    for (let i = 0; i < authChain.length; i++) {
+      headers[this.AUTH_CHAIN_HEADER_PREFIX + i] = JSON.stringify(authChain[i])
+    }
+    return headers
+  }
+
+  /**
+   * Updates or inserts an item.
+   * @param item - The item to insert or update.
+   * @param newContent - The content to be added or updated in the item. This content must be contained in the items contents.
+   */
+  async upsertItem(
+    item: LocalItem,
+    newContent: Record<string, Blob | Buffer>
+  ): Promise<RemoteItem> {
+    const contentIsContainedInItem = Object.keys(newContent).every((key) =>
+      Object.prototype.hasOwnProperty.call(item.contents, key)
```

Comment:
Any reason why we're not using `in` or just `item.contents.hasOwnProperty` ?

Reply:
That's because there's a rule in the linter that prohibits us:
`
Do not access Object.prototype method 'hasOwnProperty' from target object.eslintno-prototype-builtins
`
But I'll replace it with `in`.